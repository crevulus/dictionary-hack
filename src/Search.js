import React, { Component } from 'react';

class Search extends Component {
  constructor(props) {
    super(props);
    this.state = {
      searchValue: ''
    }
    this.handleChange = this.handleChange.bind(this);
  }

  handleChange = (event) => {
    this.setState({ searchValue: event.target.value });
  }

  handleSubmit = (event) => {
    event.preventDefault();
    this.setState({
      searchValue: ''
    });
  }

  fetchData = (searchValue) => {
    const that = this;
    fetch(`https://wordsapiv1.p.rapidapi.com/words/${this.state.searchValue}`, {
      method: "GET",
      withCredentials: true,
      headers: {
        'x-rapidapi-key': 'a2c92537d6mshed3836357d1a824p120053jsn7c815bcc2e2d'
      }
    })
      .then(resp => resp.json())
      .then(function(data) {
        that.setState({ word: data.word });
        that.setState({ pronunciation: data.pronunciation.all });
        that.setState({ category: data.results[0].partOfSpeech });
        that.setState({ definition: data.results[0].definition });
        console.log(data);
      })
      .catch(function(error) {
        console.log(error);
      });
  }

  render() {
    return (
      <form onSubmit={this.handleSubmit}>
        <label>
          Search for a word:
          <input type="text" value={this.state.searchValue} onChange={this.handleChange} />
        </label>
        <input type="submit" value="Search" onClick={this.fetchData} />
      </form>
    );
  }
}

export default Search;