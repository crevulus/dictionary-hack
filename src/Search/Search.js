import React, { Component } from 'react';
import './Search.css'

class Search extends Component {
  constructor(props) {
    super(props);
    this.state = {
      searchValue: ''
    }
  }

  render() {
    return (
      <form onSubmit={this.handleSubmit}>
        <label>
          Search for a word:
          <input type="text" value={this.state.searchValue} onChange={this.handleChange} />
        </label>
        <input type="submit" value="Search" onClick={this.fetchData} />
      </form>
    );
  }
}

export default Search;