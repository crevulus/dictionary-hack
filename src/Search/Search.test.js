/* eslint-disable */

import React from 'react';
import ReactDOM from 'react-dom';
import { render } from '@testing-library/react';
import Search from './Search';
import renderer from 'react-test-renderer';

test('renders component', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Search></Search>, div);
});

test('matches snapshot', () => {
  const snap = renderer.create(<Search></Search>).toJSON();
  expect(snap).toMatchSnapshot();
});