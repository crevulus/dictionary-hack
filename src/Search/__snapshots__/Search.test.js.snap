// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<form
  onSubmit={[Function]}
>
  <label>
    Search for a word:
    <input
      onChange={[Function]}
      type="text"
      value=""
    />
  </label>
  <input
    onClick={[Function]}
    type="submit"
    value="Search"
  />
</form>
`;
